helix-core/src/main/java/org/apache/helix/task/JobContext.java

```diff
@@ -281,9 +324,9 @@ public String getName() {
    * @param t
    */
   public void setExecutionStartTime(long t) {
-    String tStr = _record.getSimpleField(ContextProperties.EXECUTION_START_TIME.toString());
-    if (tStr == null) {
+    if (getExecutionStartTime() == WorkflowContext.NOT_STARTED) {
```

Comment:
Is there a behavior change involved here?

Reply:
Basically, getExecutionStartTime() does the same this that I removed. getExecutionStartTime() returns No_STARTED if _record.getSimpleField(ContextProperties.EXECUTION_START_TIME.toString()) is null. I used this method for simplicity and better style.

Reply:
Also, I followed the comment above this method that says "Only set the execution start time when it hasn't already been set."

Reply:
The may cause issue? I cannot remember the details. But would there be a scenario, the job is retrying and execution time has been setup before just need another update?

Reply:
Thanks for the review. Since we do not have a retry behavior for jobs and since here we are not changing the behavior, this change should be fine. Also, I just searched through the code base and didn't find such a case. We can also refer to the comment above this method,

>   /**
   * Only set the execution start time when it hasn't already been set.
   * NOTE: This method is not thread-safe. However, it is okay because even if this does get written
   * twice due to a race condition, that means the timestamps will be close enough to get a fairly
   * good estimate for the execution start time. We do not want to affect the task status update
   * performance and ultimately, this execution start time is an estimate in and of itself anyways.
   * @param t
   */

Anyways, I think it is pretty safe to do this change.